pex/installer.py

```diff
@@ -37,7 +37,10 @@ def __init__(self, source_dir, interpreter=None, install_dir=None):
     self._source_dir = source_dir
     self._install_tmp = install_dir or safe_mkdtemp()
     self._installed = None
-    self._interpreter = interpreter or PythonInterpreter.get()
+
+    from pex import vendor
+    self._interpreter = vendor.setup_interpreter(distributions=self.mixins,
```

Comment:
It feels vaguely sideeffecty to do this here... but I suppose that doing it outside has been errorprone so far.

Reply:
the `setup_interpreter` function returns a new PythonInterpreter with the same binary path but added extras - it's a bad name - no side-effects.